pkg/scheduler/core/generic_scheduler.go

```diff
@@ -1002,13 +1016,18 @@ func (g *genericScheduler) selectNodesForPreemption(
 	meta := metadataProducer(pod, nodeNameToInfo)
 	checkNode := func(i int) {
 		nodeName := potentialNodes[i].Name
+		if nodeNameToInfo[nodeName] == nil {
+			return
+		}
+		nodeInfoCopy := nodeNameToInfo[nodeName].Clone()
```

Comment:
You mean why move it? or why copy at all? if the former, I wanted to make sure all instances of copy (which is nodeinfo/meta/state) are done in the same place, if the latter, this is needed because nodeInfo gets modified during eviction evaluation inside selectVictimsOnNode

Reply:
Ok, I realize now that this was moved from inside the other function. Sg.